src/client.py

```diff
@@ -218,13 +220,48 @@ def _parse_items(self, resp_content):
         results = []
         xml = ET.ElementTree(ET.fromstring(resp_content))
         items = xml.getroot().findall("channel/item")
-        log.info(f"Found {len(items)} items from response")
+        log.info(f"Found {len(items)} items from response!")
+
         for item in items:
             result = self._parse_item(item)
             if result is not None:
                 results.append(result)
 
         return results
+    
+
+    # if we didn't get a magnet uri, attempt to resolve the magnet uri.
+    # todo for some reason Elementum cannot resolve the link that gets proxied through Jackett.
+    #  So we will resolve it manually for Elementum for now.
+    #  In actuality, this should be fixed within Elementum
+    def asinc_magnet_resolve(self, results, p_dialog):
+        size = len(results)
+        prog_from, prog_to = 20, 90
+        p_dialog.update(prog_from, message = translation(32751).format(size))
+
+        failed, count = 0, 0
+        with concurrent.futures.ThreadPoolExecutor(max_workers= os.cpu_count()*10) as executor:
```

Comment:
I'm a little confused why this needs a threadpool? Resolving magnets isn't such a heavy process to need to complexity by parallelizing.

Reply:
FYI: dew to GIL python has only concurrent threading. Which is useless in case heavy calculation, but good for IO processing.  

In this case network roundtrip is much longer then any common data processing. 20*roundrip_time makes magnet resolving last for 10 or more seconds. 
Running 20 request in parallel makes 20 request time == 1 longest request time. 

hope i explained well

Reply:
done

Reply:
Ah, yeah. I forgot this goes to the internet to get the torrent file! :sweat_smile: